tutorons/common/dblogger.py

```diff
@@ -0,0 +1,53 @@
+from tutorons.common.models import Block, ServerQuery
+from tutorons.common.htmltools import get_css_selector
+import datetime
+
+def extract_HTTP_request_fields(request):
+        url = request.POST.get('origin')
+        path = request.path_info
+        ip = request.META.get('HTTP_X_FORWARDED_FOR')
+        if ip:
+            ip = ip.split(',')[-1].strip()
+        else:
+            ip = request.META.get('REMOTE_ADDR')
+        return url, path, ip
+
+class DBLogger():
+
+    def log_query(self, request):
+        # extract request fields
+        url, path, ip = extract_HTTP_request_fields(request)
+        q = ServerQuery(ip_addr=ip, path=path)
```

Comment:
Even better: `ServerQuery.objects.create`... --- performs both the creation of the record and the saving within a single step.